src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomTerrainGenerator.java

```diff
@@ -111,8 +113,17 @@ private CustomTerrainGenerator(World world, BiomeProvider biomeProvider, CustomG
             populators.put(biome, cubicBiome.getDecorator(conf));
         }
 
-        this.strongholds = new CubicStrongholdGenerator(conf);
-        this.ravineGenerator = new CubicRavineGenerator(conf);
+        InitCubicStructuteGeneratorEvent caveEvent = new InitCubicStructuteGeneratorEvent(EventType.CAVE, new CubicCaveGenerator());
+        InitCubicStructuteGeneratorEvent strongholdsEvent = new InitCubicStructuteGeneratorEvent(EventType.STRONGHOLD, new CubicStrongholdGenerator(conf));
+        InitCubicStructuteGeneratorEvent ravineEvent = new InitCubicStructuteGeneratorEvent(EventType.RAVINE, new CubicRavineGenerator(conf));
+
+        MinecraftForge.TERRAIN_GEN_BUS.post(caveEvent);
+        MinecraftForge.TERRAIN_GEN_BUS.post(strongholdsEvent);
+        MinecraftForge.TERRAIN_GEN_BUS.post(ravineEvent);
+        
+        this.caveGenerator = (CubicCaveGenerator) caveEvent.getNewGen();
+        this.strongholds = (CubicFeatureGenerator) strongholdsEvent.getNewGen();
```

Comment:
Does it need to be this specific type? Those classes aren't in API so there is no way to actually replace them.

Reply:
Strongholds generator call `generateStructure(..)` and `getClosestStrongholdPos(...)`, which is not presented in `CubicStructureGenerator.class`. I can move those functions to `CubicStructureGenerator.class` which is moved to CC API.

Reply:
This doesn't appear to atch method names in 1.12.x vanilla. Can you see how it's done in vanilla?

Reply:
Also looks like CubicFeatureGenerator may need to be in API.

Reply:
@Barteks2x 
In Vanilla it looks like this:
```            caveGenerator = net.minecraftforge.event.terraingen.TerrainGen.getModdedMapGen(caveGenerator, net.minecraftforge.event.terraingen.InitMapGenEvent.EventType.CAVE);```
https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/patches/minecraft/net/minecraft/world/gen/ChunkGeneratorOverworld.java.patch#L8

Reply:
I wasn't asking what the event looks like, but the code for `getClosestStrongholdPos`. That method got renamed in some version.

Reply:
```            return this.strongholdGenerator.getNearestStructurePos(worldIn, position, findUnexplored);```
While `getNearestStructurePos(...)` is a member of `MapGenStructure.class`

Reply:
So rename it in `CubicFeatureGenerator` to match vanilla and move this class to API.